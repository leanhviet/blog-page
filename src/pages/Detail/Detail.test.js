// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { DetailBase } from './index'

const props = {
  blogData: {
    blog: {
      content: 'This is content',
      title: 'Title',
      createdAt: '',
    },
  },
  match: {
    params: {
      id: '1',
    },
  },
}

describe('DetailBase component', () => {
  const wrapper = shallow(<DetailBase />)
  const wrapperWithProps = shallow(<DetailBase {...props} />)

  it('should be defined DetailBase component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot DetailBase component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // Testing default props
  it('should be render defaultProps DetailBase component', () => {
    expect(DetailBase.defaultProps.blogData).toEqual({
      blog: {
        title: '',
        content: '',
        createdAt: '',
      },
    })
    expect(DetailBase.defaultProps.match).toEqual({
      params: {
        id: '1',
      },
    })
  })

  // Testing mock props
  it('should be render mockProps DetailBase component', () => {
    expect(wrapperWithProps.find('h1').props().children).toEqual(
      props.blogData.blog.title,
    )
    expect(wrapperWithProps.find('.detail__content').length).toBe(1)
  })
})
