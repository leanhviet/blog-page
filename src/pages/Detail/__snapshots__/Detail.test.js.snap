// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailBase component should be render snapshot DetailBase component 1`] = `
<Container
  className="border detail"
  fluid={false}
>
  <h1
    className="mt-4"
  />
  <p
    className="lead"
  >
    by
     
    <a
      href="/"
    >
      Start Bootstrap
    </a>
  </p>
  <hr />
  <p>
    Posted on
    Invalid date
  </p>
  <hr />
  <img
    alt=""
    className="img-fluid rounded"
    src="http://placehold.it/900x300"
  />
  <hr />
  <p
    className="lead detail__content"
  />
</Container>
`;

exports[`DetailBase component should be render snapshot DetailBase component 2`] = `
<Container
  className="border detail"
  fluid={false}
>
  <h1
    className="mt-4"
  >
    Title
  </h1>
  <p
    className="lead"
  >
    by
     
    <a
      href="/"
    >
      Start Bootstrap
    </a>
  </p>
  <hr />
  <p>
    Posted on
    Invalid date
  </p>
  <hr />
  <img
    alt="Title"
    className="img-fluid rounded"
    src="http://placehold.it/900x300"
  />
  <hr />
  <p
    className="lead detail__content"
  >
    This is content
  </p>
</Container>
`;
